modules/core/api/module.go

```diff
@@ -45,11 +45,29 @@ type IBCModule interface {
 	// OnAcknowledgementPacket is executed when a packet gets acknowledged
 	OnAcknowledgementPacket(
 		ctx context.Context,
-		sourceChannel string,
-		destinationChannel string,
+		sourceClient string,
+		destinationClient string,
 		sequence uint64,
 		acknowledgement []byte,
 		payload channeltypesv2.Payload,
 		relayer sdk.AccAddress,
 	) error
 }
+
+type WriteAcknowledgementWrapper interface {
+	// WriteAcknowledgement writes the acknowledgement for an async acknowledgement
+	WriteAcknowledgement(
+		ctx context.Context,
+		clientID string,
+		sequence uint64,
+		ack channeltypesv2.Acknowledgement,
+	) error
+}
+
+// PacketDataUnmarshaler defines an optional interface which allows a middleware
+// to request the packet data to be unmarshaled by the base application.
+type PacketDataUnmarshaler interface {
```

Comment:
Does this need to carry the same name as the old interface? We can call it `PayloadUnmarshaler`